src/py-opentimelineio/opentimelineio/adapters/otioz.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright 2019 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""OTIOZ adapter - bundles otio files linked to local media
+
+Takes as input an OTIO file that has media references which are all relative
+local paths (ie file:///foo.mov) and bundles those files and the otio file into
+a single zip file with the suffix .otioz.  Can error out if files aren't
+locally referenced or provide missing references
+
+Can also extract the content.otio file from an otioz bundle for processing.
+"""
+
+import os
+import copy
+import zipfile
+
+from .. import (
+    exceptions,
+    schema,
+)
+
+from . import (
+    file_bundle_utils as utils,
+    otio_json
+)
+
+
+def read_from_file(filepath, extract_to_directory=None):
+    if not zipfile.is_zipfile(filepath):
+        raise exceptions.OTIOError("Not a zipfile: {}".format(filepath))
+
+    if extract_to_directory:
+        output_media_directory = os.path.join(
+            extract_to_directory,
+            utils.BUNDLE_DIR_NAME
+        )
+
+        if not os.path.exists(extract_to_directory):
+            raise exceptions.OTIOError(
+                "Directory '{}' does not exist, cannot unpack otioz "
+                "there.".format(extract_to_directory)
+            )
+
+        if os.path.exists(output_media_directory):
+            raise exceptions.OTIOError(
+                "Error: '{}' already exists on disk, cannot overwrite while "
+                " unpacking OTIOZ file '{}'.".format(
+                    output_media_directory,
+                    filepath
+                )
+
+            )
+
+    with zipfile.ZipFile(filepath, 'r') as zi:
+        result = otio_json.read_from_string(zi.read(utils.BUNDLE_PLAYLIST_PATH))
+
+        if extract_to_directory:
+            zi.extractall(extract_to_directory)
+
```

Comment:
Would it make sense to relink all the media_reference `target_url` attributes here when `extract_to_directory` is set?

Reply:
you mean with an absolute path?  I think they get set to relative paths as is.  But thats a good question.

Reply:
I like the idea of using relative paths to point to stuff in the media folder. This will help to move forward some strictness and uses of the target_url which are too loose at the moment.

Reply:
Makes sense. As long as the paths are relative to the `.otio` file then you can move the `.otiod` around on your filesystem and have everything still be properly pathed.